agent/netif.go

```diff
@@ -45,7 +45,7 @@ func (netif *NetIf) UnmarshalJSON(data []byte) error {
 	m := make(map[string]string)
```

Comment:
This is needed because we use 
// a type like net.IP here, not a simple type, and so a call to
// net.ParseIP is required to unmarshal this properly.